app/src/forms/common/scheduleService.js

```diff
@@ -0,0 +1,597 @@
+ * Check if a date is valid
+ * @param {String|Date} date The date to validate
+ * @returns {Boolean} True if the date is valid
+ */
+function isDateValid(date) {
+  return date && !isNaN(Date.parse(date));
+}
+
+/**
+ * Check if two dates are the same day
+ * @param {String|Date} date1 First date
+ * @param {String|Date} date2 Second date
+ * @param {String} timezone The timezone to use
+ * @param {Boolean} compareTime Whether to compare exact time (false = day only)
+ * @returns {Boolean} True if dates are the same (day or exact time based on compareTime)
+ */
+function isSameDay(date1, date2, timezone = 'America/Vancouver', compareTime = false) {
+  if (!date1 || !date2) return false;
+
+  const d1 = moment.tz(date1, timezone);
+  const d2 = moment.tz(date2, timezone);
+
+  if (compareTime) {
+    return d1.isSame(d2);
+  } else {
+    return d1.startOf('day').isSame(d2.startOf('day'));
+  }
+}
+
+/**
+ * Calculate days between two dates
+ * @param {String|Date} startDate Start date
+ * @param {String|Date} endDate End date
+ * @param {String} timezone The timezone to use
+ * @param {Boolean} ignoreTime Whether to ignore time component (defaults to true)
+ * @returns {Number} Number of days between the dates
+ */
+function daysBetween(startDate, endDate, timezone = 'America/Vancouver', ignoreTime = true) {
+  if (!startDate || !endDate) return 0;
+
+  let start = moment.tz(startDate, timezone);
+  let end = moment.tz(endDate, timezone);
+
+  if (ignoreTime) {
+    start = start.startOf('day');
+    end = end.startOf('day');
+  }
+
+  return end.diff(start, 'days');
+}
+
+/**
+ * Schedule-specific functions
+ * These work with schedule objects by extracting parameters and
+ * passing them to the core date functions
+ */
+
+/**
+ * @function checkIsFormExpired
+ * Checks if a form is expired based on its schedule settings
+ * @param {Object} formSchedule - The form's schedule configuration
+ * @returns {Object} {allowLateSubmissions: Boolean, expire: Boolean, message: String}
+ */
+const checkIsFormExpired = (formSchedule = {}) => {
+  // Default result
+  let result = {
+    allowLateSubmissions: false,
+    expire: false,
+    message: '',
+  };
+
+  // Early exit if scheduling is not enabled or no opening date is set
+  if (!formSchedule || !formSchedule.enabled || !formSchedule.openSubmissionDateTime) {
+    return result;
+  }
+
+  const currentMoment = moment.utc();
+
+  // Apply defaults for missing fields
+  const defaults = {
+    openSubmissionTime: formSchedule.openSubmissionTime || '00:00',
+    closeSubmissionTime: formSchedule.closeSubmissionTime || (formSchedule.closeSubmissionDateTime ? '23:59' : null),
+    timezone: formSchedule.timezone || 'America/Vancouver',
+  };
+  const schedule = { ...formSchedule, ...defaults };
+
+  // Opening Time
+  const openDateTime = `${schedule.openSubmissionDateTime} ${schedule.openSubmissionTime}`;
+  const openingMoment = moment.tz(openDateTime, 'YYYY-MM-DD HH:mm', schedule.timezone).utc();
```

Comment:
why do we have utc in these calculations but not everywhere? do we need utc is we are setting the timezone?